tests/test_config_entries.py

```diff
@@ -4323,3 +4323,174 @@ async def test_hashable_non_string_unique_id(
     del entries[entry.entry_id]
     assert not entries
     assert entries.get_entry_by_domain_and_unique_id("test", unique_id) is None
+
+
+async def test_avoid_starting_config_flow_on_single_instance(
+    hass: HomeAssistant, manager: config_entries.ConfigEntries
+) -> None:
+    """Test that we cannot start a config flow for a single instance only integration that already has an entry."""
```

Comment:
Please limit docstring lines to 72 characters per line. The first line is a header and must be a single sentence.